x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/device.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fetcher
+
+import (
+	"github.com/google/uuid"
+
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/entityanalytics/internal/collections"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// TODO: Implement fetchers for the registered owners and users
+// for devices. These will then be retained in the following fields
+// of Device.
+//
+// 	// A set of UUIDs for registered owners of this device.
+// 	RegisteredOwners collections.UUIDSet `json:"registeredOwners"`
+// 	// A set of UUIDs for registered users of this device.
+// 	RegisteredUsers collections.UUIDSet `json:"registeredUsers"`
+//
+// and the addition of the following lines to Merge
+//
+// 	other.RegisteredOwners.ForEach(func(elem uuid.UUID) {
+// 		d.RegisteredOwners.Add(elem)
+// 	})
+// 	other.RegisteredUsers.ForEach(func(elem uuid.UUID) {
+// 		d.RegisteredUsers.Add(elem)
+// 	})
+//
+// with associated test extensions.
+
+// Device represents a device identity asset.
+type Device struct {
+	// The ID (UUIDv4) of the device.
+	ID uuid.UUID `json:"id"`
+	// The attributes for the device.
+	Fields mapstr.M `json:"fields"`
+	// A set of UUIDs which are groups this device is a member of.
+	MemberOf collections.UUIDSet `json:"memberOf"`
+	// A set of UUIDs which are groups this device is a transitive member of.
+	TransitiveMemberOf collections.UUIDSet `json:"transitiveMemberOf"`
+	// Discovered indicates that this device was newly discovered. This does not
+	// necessarily imply the device was recently added in Azure Active Directory,
+	// but it does indicate that it's the first time the device has been seen by
+	// the input.
+	Discovered bool `json:"-"`
+	// Modified indicates that an attribute or group membership has been
+	// modified on this device.
+	Modified bool `json:"-"`
+	// Deleted indicates the device has been deleted.
+	Deleted bool `json:"deleted"`
```

Comment:
Are `Discovered` `Modified` `Deleted` Mutually exclusive?

Reply:
Yes. I was thinking of using an enum, but this is what was already being used, so I retained the same approach.